cmake/modules/hunter_install_libsodium_config.cmake

```diff
@@ -0,0 +1,20 @@
+include(CMakePackageConfigHelpers)
+include(hunter_test_string_not_empty)
+
+function(hunter_install_libsodium_config)
+
+    hunter_test_string_not_empty("${HUNTER_libsodium_VERSION}")
+    hunter_test_string_not_empty("${HUNTER_PACKAGE_INSTALL_PREFIX}")
+    hunter_test_string_not_empty("${HUNTER_SELF}")
+
+    set(templates "${HUNTER_SELF}/cmake/templates")
+    set(configs "${HUNTER_PACKAGE_INSTALL_PREFIX}/lib/cmake/libsodium")
+
+    write_basic_package_version_file("${configs}/libsodiumConfig.cmake" VERSION "${HUNTER_libsodium_VERSION}" COMPATIBILITY ExactVersion)
+
+    configure_file(
+            "${templates}/libsodiumConfig.cmake.in"
+            "${configs}/libsodiumConfig.cmake"
```

Comment:
they will indeed overwrite eachother but I've added switched so it its a debug build it will use the debug library (for MSBuild at least, not sure what to do with the whole autotools situation not doing the right thing with HUNTER_PACKAGE_CONFIGURATION_TYPES) Is it OK like this or should I take out and split the targets according to build config?